pandajedi/jedidog/AtlasDataLocalityUpdater.py

```diff
@@ -0,0 +1,155 @@
+                                                            pid=self.pid, timeLimit=720)
+            if not got_lock:
+                tmpLog.debug('locked by another process. Skipped')
+                return
+            # get list of datasets
+            datasets_list = self.get_datasets_list()
+            # make thread pool
+            thread_pool = ThreadPool()
+            # make workers
+            n_workers = 4
+            for _ in range(n_workers):
+                thr = ContentsFeederThread( taskDsList=datasets_list,
+                                            threadPool=thread_pool,
+                                            taskbufferIF=self.taskBufferIF,
+                                            ddmIF=self.ddmIF,
+                                            pid=self.pid,
+                                            loggerObj=tmpLog)
+                thr.start()
+            # join
+            thread_pool.join()
+            # done
+            tmpLog.debug('done')
+        except Exception:
+            errtype, errvalue = sys.exc_info()[:2]
+            tmpLog.error('failed with {0} {1} {2}'.format(errtype, errvalue, traceback.format_exc()))
+
+    # clean up old data locality records in DB table
+    def doCleanDataLocality(self):
+        try:
+            tmpLog = MsgWrapper(logger, ' #ATM #KV doUpdateDataLocality')
+            tmpLog.debug('start')
+            # lock
+            got_lock = self.taskBufferIF.lockProcess_JEDI(  vo=self.vo, prodSourceLabel=self.prodSourceLabel,
+                                                            cloud=None, workqueue_id=None, resource_name=None,
+                                                            component='AtlasDataLocalityUpdaterWatchDog.doCleanDataLocality',
+                                                            pid=self.pid, timeLimit=1440)
+            if not got_lock:
+                tmpLog.debug('locked by another process. Skipped')
+                return
+            # lifetime of records
+            record_lifetime_hours = 72
+            # run
+            now_timestamp = datetime.datetime.utcnow()
+            before_timestamp = now_timestamp - datetime.timedelta(hours=record_lifetime_hours)
+            n_rows = self.taskBufferIF.deleteOutdatedDatasetLocality_JEDI(before_timestamp)
+            tmpLog.info('cleaned up {0} records'.format(n_rows))
+            # done
+            tmpLog.debug('done')
+        except Exception:
+            errtype, errvalue = sys.exc_info()[:2]
+            tmpLog.error('failed with {0} {1} {2}'.format(errtype, errvalue, traceback.format_exc()))
+
+    # main
+    def doAction(self):
+        try:
+            # get logger
+            origTmpLog = MsgWrapper(logger)
+            origTmpLog.debug('start')
+            # clean up data locality
+            self.doCleanDataLocality()
+            # update data locality
+            self.doUpdateDataLocality()
```

Comment:
I added another watchdog AtlasDataLocalityUpdaterWatchDog to update and clean up JEDI_DATASET_LOCALITY table.

If it is okay to, I will modify AtlasTaskWithholderWatchDog accordingly.